### DISCUSSION TITLE
Monte Carlo model - lattice

### DISCUSSION THREAD

**Discussion by MajaR01** [Part-II-Computational-Physics/.github#62] [General]
Hi, I am a bit confused in the first part of the exercise about the variable 'lattice' - is it init_lattice? And if so, how should I rename it, so I can use it and be able to set different values of width and type instead of just hard coding them at the beginning?
Here is my code in its current state:
width =2
type=1
def init_lattice(width,type):
    '''Produce an initial lattice with spins. Type 1 gives all spins 1, type 0 gives a random spins, type -1 gives all spins -1. Captures invalid types.'''
    if type ==1:
        lattice = np.ones((width,width),dtype=int)
    elif type ==0:
        lattice = rng.random((width,width))
    elif type == -1:
        lattice = (-1)*np.ones((width,width),dtype=int)
    else:
        print('Invalid type of a lattice')
    return lattice
#lattice = init_lattice(width,type)    
def neighbouring_sites(i,j,width):
    '''Return the coordinates of the 4 sites adjacent to [i,j] on an width*width lattice. Takes into account periodic boundary conditions.'''
    return [((i+1)%width,j),((i-1)%width,j),(i,(j+1)%width),(i,(j-1)%width)]

def neighbouring_spins_sum(i,j,lattice,width):
    '''Sums the spins of all neighbours of the spin at [i,j].'''
    return np.sum(sum(row) for row in neighbouring_sites(i,j,width))

def compute_magnetisation(lattice):
    '''Computes the magnetisation of the lattice.'''
    return np.sum(init_lattice)

def plot_lattice(lattice,ax,title):
    '''Plot the lattice configuration.'''
    ax.matshow(lattice, vmin=-1, vmax=1, cmap=plt.cm.binary)
    ax.title.set_text(title)
    ax.set_yticklabels([])
    ax.set_xticklabels([])
    ax.set_yticks([])
    ax.set_xticks([])
#init_lattice(2,1)
#neighbouring_sites(1,2,2)
#neighbouring_spins_sum(1,2,init_lattice(2,1),width)

**Comment by DLHvdH** [comment]
Hi Maya, I'm not sure if I understand your question correctly. But the idea is that the function init_lattice creates and initialises a lattice to be used for the Monte Carlo simulations.

So you'd use it like 

```
w = 25
t = 1
lattice = init_lattice(w,t)
```

to get at 25x25 lattice with all spins up (+1).
So later you can, for example, call 
`Magnetisation = compute_magnetisation(lattice)`

Also, you don't have to wait until the end of the if-else statements before returning the lattice.
You can also do things like
```
if type ==1:
    return np.ones((width,width),dtype=int)
```
